src/main/java/com/example/training/service/AccidentsService.java

```diff
@@ -19,28 +20,18 @@ public AccidentsService(AccidentsProxyService accidentsProxyService) {
     }
 
     public List<AccidentDto> getAccidentsByPolicy(String policyId) {
-        var accidents = accidentsProxyService.getAccidentsByPolicy(policyId);
-        return mapToAccidentDto(accidents);
+        Accident[] accidents = accidentsProxyService.getAccidentsByPolicy(policyId);
+        List<Accident> accidentList = Arrays.asList(accidents);
+        return AccidentMapper.INSTANCE.accidentsToAccidentDtos(accidentList);
     }
 
     public AccidentDto getAccidentById(String accidentId) {
         var accident = accidentsProxyService.getAccidentById(accidentId);
-        return accident != null ? new AccidentDto(accident.getAccidentId(), accident.getDescription(), accident.getStatus()) : null;
-    }
-
-    private List<AccidentDto> mapToAccidentDto(Accident[] accidents) {
-        return accidents != null ?
-                List.of(accidents).stream()
-                        .map(accident -> new AccidentDto(accident.getAccidentId(), accident.getStatus()))
-                        .collect(Collectors.toList()) :
-                List.of();
+        return accident != null ? AccidentMapper.INSTANCE.accidentToAccidentDto(accident) : null;
```

Comment:
This means your service can return a null... not the best API, nulls are evil and hard to manage. Is it "normal" to get here with a non existent accidentId? It would be better to just break the execution and throw an empty data access exception in order to avoid having to deal with undesired results in the controller